### PR TITLE
Store peerbook address keys reversed for better random distribution

### PR DESCRIPTION
**Pull request by Vagabond**
Storing peer address keys with the type bytes at the start leads to
extremely bad distribution of the keys which makes the random function
very likely to select the same peers over and over.

This patch stores the keys reversed so the tag bytes are the least
significant ones. The peerbook is rewritten on startup if needed.

### CODE REVIEW

File: src/peerbook/libp2p_peerbook.erl [hunk 1]
```diff
@@ -185,7 +185,7 @@ random(Peerbook=#peerbook{store=Store, stale_time=StaleTime}, Exclude, Pred, Tri
                 false ->
```
Comment (evanmcc) [inline src/peerbook/libp2p_peerbook.erl:185]: map rev over the exclude list or it won't match right, I think

**Review by evanmcc — COMMENTED** [review]
other than the one comment this looks good.  I'm concerned about the rewrite at the beginning, I'd rather just drop it, but this should work too.

File: src/peerbook/libp2p_peerbook.erl (hunk 1 above)
Comment (Vagabond) [reply src/peerbook/libp2p_peerbook.erl:185]: good catch!

Comment (Vagabond) [reply src/peerbook/libp2p_peerbook.erl:186]: done

**Comment by Vagabond** [comment]
> other than the one comment this looks good. I'm concerned about the rewrite at the beginning, I'd rather just drop it, but this should work too.

I tested the rewrite on 1gb ram hotspots and it went fine.

**Review by madninja — APPROVED** [review]

**Review by evanmcc — APPROVED** [review]

**Pull request merged**